### PR TITLE
[Config]: Add key_features

### PR DESCRIPTION
**Pull request by meatball133**
@pfertyk, this will be my first pr, I am still busy working on some other tracks but I really want to help you out with this track. I had some time so I thought why not list the key features.

These looks like this on the website:

image

I used this [link](https://docs.godotengine.org/en/stable/about/faq.html#what-were-the-motivations-behind-creating-gdscript) as inspiration. 

But with that said you might find some features are more important than the once I listed. If that is the case just tell me and we can update it, I am not stuck with these or anything.

The icons can be found at the bottom of this page: https://exercism.org/docs/building/tracks/icons

### CODE REVIEW

**Comment by pfertyk** [comment]
Wow, that's a very nice collection, thanks :) These are good, I will take a closer look a bit later to see if maybe I could add anything to the list. I was thinking about the GD Native feature (which is running other languages in Godot, most notably C# and C++, but also Python, Rust etc), but that seem to be more of a Godot Engine feature than GDScript feature. What do you think?

**Comment by meatball133** [comment]
We can only have 6 of these "key_features", I am open to removing some of the currents once if you feel like something else is more important is what I meant.

The c++ bindings I listed may actually be bindings to the engine and not gdscript. I am personally not a c++ developer. So never tried anything of that.

What should be listed are language features, not features of libraries/tools. Thereby that c++ binding will likely need to be removed and replaced with something else. Some languages have compatibility working with other languages, c bindings are really common. So I would personally not recommend talking about other languages if it isn't the language itself that can work with them.

**Comment by pfertyk** [comment]
Good point. In Godot, the native binding is done using plugins, like [PythonScript](https://godotengine.org/asset-library/asset/179). Also, the native code is then used instead of GDScript, so I think this shouldn't be listed.

I would like to suggest asking the PM of Godot Engine, Rémi Verschelde, to take a look at this. He has extensive knowledge about the features of the language and the engine, and it would be great to use his expertise and refine our list. I'm sure he wouldn't mind ;) However, I don't want to ping him if you believe that this is a bad idea (I'm don't know yet how these things are done in Exercism). What is your opinion?

**Comment by meatball133** [comment]
I think contacting Remi is a good idea, just keep in mind 6 is the maximum amount of key features. I quite often talk with crystal developers/community to get their views on how to do things, so personally, I am for bringing in outside views on a topic.

**Comment by pfertyk** [comment]
Then let me invite him ;) @akien-mga could you please help us refine the list of key features of GDScript? As mentioned above, there should be at most 6 of them ;)

File: config.json [hunk 1]
```diff
@@ -26,6 +26,37 @@
...
+    {
+      "title": "Specialized types for game development",
+      "content": "Specialized types for increased productivity and performance, when working with 2D and 3D games.",
+      "icon": "powerful"
+    },
+    {
+      "title": "Garbage collection",
+      "content": "Garbage collection is automatic and seamless, freeing you from manual memory management.",
+      "icon": "garbage-collected"
+    },
```
Comment (adamscott) [inline config.json:54]: Godot is using reference counting for memory management, not garbage collection. It's one of the features of GDScript, _not being_ garbage collected. :)

File: config.json [hunk 2]
```diff
@@ -26,6 +26,37 @@
...
+    {
+      "title": "Cross platform",
+      "content": "Gdscript code can run on Windows, Linux, macOS, iOS, Android, and the web.",
+      "icon": "cross-platform"
+    },
+    {
+      "title": "C++ bindings",
+      "content": "C++ bindings allows for fast, low-level access to the engine's features.",
+      "icon": "fast"
+    },
```
Comment (adamscott) [inline config.json:44]: I agree with @pfertyk that this isn't GDScript related, but is Godot related.

GDScript is only the scripting engine that runs on top of the Godot features.

File: config.json [hunk 3]
```diff
@@ -26,6 +26,37 @@
...
-  "key_features": [],
+  "key_features": [
+    {
+      "title": "Strong, dynamic typing",
+      "content": "Gdscript is strong and dynamically typed and supports gradual typing.",
+      "icon": "dynamically-typed"
+    },
+    {
+      "title": "Cross platform",
+      "content": "Gdscript code can run on Windows, Linux, macOS, iOS, Android, and the web.",
```
Comment (adamscott) [inline config.json:37]:
```suggestion
      "content": "GDScript code can run wherever Godot runs, either on Windows, Linux, macOS, iOS, Android, the web and even consoles.",
```

**Comment by adamscott** [comment]
I would add something like this:
```json
{
    "title": "Tightly coupled",
    "content": "GDScript has a deep integration within the Godot game engine: no installation required to use it, it just works. There's even a script editor within!",
    "icon": "secret-handshake"
}
```

**Comment by meatball133** [comment]
Yes sounds good. Although those icons are not available. You have to select one of the icons at the bottom or those with the color purple.

I think the `embeddable` icon could work, but I am open for other suggestions.

File: config.json (hunk 1 above)
Comment (adamscott) [reply config.json:54]:
```suggestion
    {
      "title": "No garbage collection",
      "content": "Garbage collection can cause random freezes during gameplay. Godot (and thus GDScript) instead use reference counting, freeing you from manual memory management.",
      "icon": "garbage-collected"
    },
``

**Comment by adamscott** [comment]
> I think the `embeddable` icon could work, but I am open for other suggestions.

Yes! I think it's the more appropriate one.

**Comment by pfertyk** [comment]
Thank you very much for your input @adamscott ! Things have been a bit slow around the GDScript track, let's pick up the pace ;)

**Review by pfertyk — COMMENTED** [review]
I've left a few small notes ;)

**Comment by pfertyk** [comment]
Many thanks @meatball133 for contributing and for all the adjustments! ;) I'm merging this for now. With key featured in our docs, we are one step closer to releasing the track! :tada: 

If you have any other feedback about key features @adamscott , please let us know ;) I would also love to get @akien-mga 's opinion on this, but I'm not sure if he can find the time :)

**Pull request merged**